osu.Game/Tests/Visual/SkinnableTestScene.cs

```diff
@@ -152,25 +153,38 @@ private class TestLegacySkin : LegacySkin
         {
             private readonly bool extrapolateAnimations;
 
+            private readonly HashSet<string> legacyFontPrefixes = new HashSet<string>();
+
             public TestLegacySkin(SkinInfo skin, IResourceStore<byte[]> storage, AudioManager audioManager, bool extrapolateAnimations)
                 : base(skin, storage, audioManager, "skin.ini")
             {
                 this.extrapolateAnimations = extrapolateAnimations;
+
+                legacyFontPrefixes.Add(GetConfig<string, string>("HitCirclePrefix")?.Value ?? "default");
+                legacyFontPrefixes.Add(GetConfig<string, string>("ScorePrefix")?.Value ?? "score");
+                legacyFontPrefixes.Add(GetConfig<string, string>("ComboPrefix")?.Value ?? "score");
             }
 
             public override Texture GetTexture(string componentName, WrapMode wrapModeS, WrapMode wrapModeT)
             {
                 // extrapolate frames to test longer animations
-                if (extrapolateAnimations)
-                {
-                    var match = Regex.Match(componentName, "-([0-9]*)");
-
-                    if (match.Length > 0 && int.TryParse(match.Groups[1].Value, out var number) && number < 60)
-                        return base.GetTexture(componentName.Replace($"-{number}", $"-{number % 2}"), wrapModeS, wrapModeT);
-                }
+                if (extrapolateAnimations && isAnimationComponent(componentName, out var number) && number < 60)
```

Comment:
where's this `number < 60` check coming from?

Reply:
not sure what you mean, `isAnimationComponent(...)` is fulfilling its job, `number < 60` is took from above.

Reply:
fair enough, not sure how I missed this